src/components/NewsCardList/NewsCardList.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import NewsCard from "../NewsCard/NewsCard";
+
+function NewsCardList(props) {
+  const [articlesToRender, setArticlesToRender] = useState(3);
+
+  const handleMoreNewsClick = () => {
+    if (articlesToRender < props.searchResults.length) {
+      localStorage.setItem("counter", parseInt(articlesToRender) + 3);
+      setArticlesToRender(localStorage.getItem("counter"));
+    }
+  };
+
+  return (
+    <section
+      className={
+        props.isSearchResult ? "cards-list cards-list_active" : "cards-list"
+      }
+    >
+      <h2 className="cards-list__title">Search results</h2>
+      <ul className="cards-list__container">
+        {props.searchResults
+          .slice(0, articlesToRender)
+          .map((article, index) => (
+            <NewsCard
+              key={"articleId_" + Math.round(Math.random() * 555555)}
```

Comment:
`              key={"articleId_" + Math.round(Math.random() * 555555)}
`

no need to use `Math.round(Math.random() * 555555` because then it's better to use `index`. But it’s better to use `_id` of the card instead of the `index` for attribute `key` (please, make sure that you don't duplicate articles ). 




Here is why we shouldn’t use `index` for that: [click here]( https://programmingwithmosh.com/react/why-do-i-need-keys-in-react-lists/#:~:text=React%20recommends%20that%20you%20do,to%20some%20unstable%20component%20behaviour.)